CMakeLists.txt

```diff
@@ -27,6 +27,78 @@ ELSE()
     SET(VER_BUILD 0)
 ENDIF()
 
+# Configuration of compilers
+## Compiler flags to add
+### Flags for GCC and Clang
+set(GNU_FLAGS_WARNING_YES "-Wall -Wextra -Wformat=2 -Wpedantic -Wshadow -Wmissing-declarations -Wfloat-equal")
+set(GNU_FLAGS_WARNING_NO  "-Wno-unused-command-line-argument -Wno-documentation -Wno-format-nonliteral")
+set(GNU_FLAGS_WARNING     "${GNU_FLAGS_WARNING_YES} ${GNU_FLAGS_WARNING_NO}")
+set(GNU_FLAGS_ERROR       "-pedantic-errors -Werror") # warning is error
```

Comment:
@RyDroid Well, here, I dislike pedantic errors. Usually, they don't always make sense and bother a lot.